yamux/src/frame.rs

```diff
@@ -105,52 +160,68 @@ impl Frame<Data> {
             header.ack()
         }
 
-        Frame::new(header)
+        Frame::from_header(header)
     }
+}
 
-    pub fn body(&self) -> &[u8] {
-        &self.body
+impl Frame<WindowUpdate> {
+    pub fn window_update(id: StreamId, credit: u32) -> Frame<WindowUpdate> {
+        Frame::from_header(Header::window_update(id, credit))
     }
+}
 
-    pub fn body_len(&self) -> u32 {
-        // Safe cast since we construct `Frame::<Data>`s only with
-        // `Vec<u8>` of length [0, u32::MAX] in `Frame::data` above.
-        self.body().len() as u32
+impl Frame<GoAway> {
```

Comment:
Can we re-order the code such that this doesn't show up as this much diff?

Reply:
ok